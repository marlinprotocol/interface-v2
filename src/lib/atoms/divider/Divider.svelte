<script lang="ts">
	import type { DividerDirection } from '$lib/types/componentTypes';
	import { cn } from '$lib/utils/helpers/commonHelper';
	import {
		getDividerColor,
		getDividerHeight,
		getDividerWidth
	} from '$lib/utils/helpers/componentHelper';

	export let direction: DividerDirection = 'divider-horizontal';
	export let margin = 'my-0';
	export let color: string = '';
	export let height: string = '';
	export let width: string = '';

	$: dividerHeight = height !== '' ? height : getDividerHeight(direction);
	$: dividerWidth = width !== '' ? width : getDividerWidth(direction);
	$: dividerColor = color !== '' ? color : getDividerColor(direction);
</script>

<div
	class={cn(dividerHeight, dividerWidth, dividerColor, margin, 'shrink-0')}
	data-testid="divider"
/>
