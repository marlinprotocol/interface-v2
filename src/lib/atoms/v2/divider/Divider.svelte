<script lang="ts">
	import type { DividerDirection } from '$lib/types/v2/componentTypes';
	import {
		getDividerColor,
		getDividerHeight,
		getDividerWidth
	} from '$lib/utils/v2/helpers/componentHelper';

	export let direction: DividerDirection = 'divider-horizontal';
	export let margin = 'my-0';
	export let color: string = '';
	export let height: string = '';
	export let width: string = '';

	$: dividerHeight = height !== '' ? height : getDividerHeight(direction);
	$: dividerWidth = width !== '' ? width : getDividerWidth(direction);
	$: dividerColor = color !== '' ? color : getDividerColor(direction);
</script>

<div class="{dividerHeight} {dividerWidth} {dividerColor} {margin}" data-testid="divider" />
