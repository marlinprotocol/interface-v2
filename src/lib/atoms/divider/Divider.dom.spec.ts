import Divider from './Divider.svelte';
import { describe, test, afterEach } from 'vitest';
import { cleanup, render } from '@testing-library/svelte';

describe('Divider', () => {
	afterEach(() => {
		cleanup();
	});

	it('renders the divider component', () => {
		const { container } = render(Divider);
		const dividerElement = container.firstChild;
		expect(dividerElement).toMatchSnapshot();
	});

	it('applies default styles based on direction', () => {
		const { getByTestId } = render(Divider, { direction: 'divider-horizontal' });
		const expectedClasses = ['bg-black/[0.1]', 'w-full', 'h-[1px]'];
		expectedClasses.forEach((expectedClass) => {
			expect(getByTestId('divider').className.split(' ')).toContain(expectedClass);
		});
	});

	it('applies vertical styles when direction is "divider-vertical"', () => {
		const { getByTestId } = render(Divider, { direction: 'divider-vertical' });
		const expectedClasses = ['bg-grey-1000', 'w-[1px]', 'h-4'];
		expectedClasses.forEach((expectedClass) => {
			expect(getByTestId('divider').className.split(' ')).toContain(expectedClass);
		});
	});

	it('overrides default color when a custom color prop is provided', () => {
		const customColor = 'bg-red-500';
		const { getByTestId } = render(Divider, { color: customColor });
		expect(getByTestId('divider').className.split(' ')).toContain(customColor);
	});

	it('overrides default width when a custom width prop is provided', () => {
		const customWidth = 'w-2';
		const { getByTestId } = render(Divider, { width: customWidth });
		expect(getByTestId('divider').className.split(' ')).toContain(customWidth);
	});

	it('overrides default height when a custom height prop is provided', () => {
		const customHeight = 'h-8';
		const { getByTestId } = render(Divider, { height: customHeight });
		expect(getByTestId('divider').className.split(' ')).toContain(customHeight);
	});

	it('applies margin classes based on the margin prop', () => {
		const customMargin = 'my-4';
		const { getByTestId } = render(Divider, { margin: customMargin });
		expect(getByTestId('divider').className.split(' ')).toContain(customMargin);
	});
});
