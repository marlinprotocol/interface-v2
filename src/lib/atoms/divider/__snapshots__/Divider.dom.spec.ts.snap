// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > renders the divider component 1`] = `
<div
  class="h-[1px] w-full bg-black/[0.1] my-0 shrink-0"
  data-testid="divider"
/>
`;
